lib/plugins/aws/provider/awsProvider.test.js

```diff
@@ -46,9 +56,67 @@ describe('AwsProvider', () => {
       const newAwsProvider = new AwsProvider(serverless, options);
 
       expect(typeof newAwsProvider.sdk.config.httpOptions.agent).to.not.equal('undefined');
+    });
+
+    it('should set AWS ca single', () => {
+      process.env.ca = '-----BEGIN CERTIFICATE-----\n-----END CERTIFICATE-----';
+      const newAwsProvider = new AwsProvider(serverless, options);
+
+      expect(typeof newAwsProvider.sdk.config.httpOptions.agent).to.not.equal('undefined');
+    });
+
+    it('should set AWS ca multiple', () => {
+      const certContents = '-----BEGIN CERTIFICATE-----\n-----END CERTIFICATE-----';
+      process.env.ca = `${certContents},${certContents}`;
+      const newAwsProvider = new AwsProvider(serverless, options);
+
+      expect(typeof newAwsProvider.sdk.config.httpOptions.agent).to.not.equal('undefined');
+    });
+
+    it('should set AWS cafile single', () => {
+      const tmpdir = os.tmpdir();
+      const certContents = '-----BEGIN CERTIFICATE-----\n-----END CERTIFICATE-----';
+      const filePath = path.join(tmpdir, 'ca.txt');
+      fs.writeFileSync(filePath, certContents);
+      process.env.cafile = filePath;
+      const newAwsProvider = new AwsProvider(serverless, options);
+
+      expect(typeof newAwsProvider.sdk.config.httpOptions.agent).to.not.equal('undefined');
 
       // clear env
```

Comment:
This cleanup should also be moved into the `afterEach` block.

Reply:
"This" meaning the comment or the unlinkSync() call?

If the latter, I'm new to Mocha so I might be wrong here but in my reading of the docs, this won't work.

Given the following:

- A hook should be defined in the describe block or at the root level.
- fs.unlink() throws an exception if the file is not found
- not all tests in the constructor's describe() block require these files so for most tests the files will not exist and an exception will be thrown

Therefore, I don't think this will work. Thoughts?

Reply:
@pmuens -- in case you didn't see this comment.

Reply:
Thanks for the reply and pinging here @geofflancaster 👍

🤔 Good catch! Yes, I think you're right. The only feasible solution to solve this problem is to wrap all the tests which rely on the linking into a `describe()` block and then add a `afterEach()` block which takes care of the unlinking (via `fs.unlinkSync()`) there.